snap/info_snap_yaml.go

```diff
@@ -70,6 +71,7 @@ func InfoFromSnapYaml(yamlData []byte) (*Info, error) {
 		Name:        y.Name,
 		Developer:   y.Developer,
 		Version:     y.Version,
+		Revision:    y.Revision,
```

Comment:
Sadly, it doesn't. I mean it *makes sense*, but it's not going to happen. Revision number is external to the thing being revisioned. You want something like `MkStoreYaml` from `snap/removed/removed_test.go`:
```go
func (s *removedSuite) MkStoreYaml(c *check.C, pkgType snap.Type) {
        // creating the part to get its manifest path is cheating, a little
        info := &snap.Info{
                Name:      "foo",
                Developer: "bar",
                Version:   "1",
                Type:      pkgType,
        }

        content := `
name: foo
origin: bar
version: 1
revision: 999999999999
type: app
description: |-
  bla bla bla
iconurl: http://i.stack.imgur.com/i8q1U.jpg
downloadsize: 5554242
`
        p := snappy.RemoteManifestPath(info)
        c.Assert(os.MkdirAll(filepath.Dir(p), 0755), check.IsNil)
        c.Assert(ioutil.WriteFile(p, []byte(content), 0644), check.IsNil)

}
```